Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/aspects/LoggingAspect.java

```diff
@@ -1,49 +1,188 @@
             throw err;
         }
     }
 
+    @Before("bean(createExamUseCase)")
+    public void logCreateExamUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var request = (CreateExamUseCase.Request) args[0];
+        CREATE_EXAM_USE_CASE_LOGGER.info("[Create Exam] name=\"{}\" startTime=\"{}\" endTime=\"{}\" description=\"{}\"",
+                request.getName(), format(request.getStartTime()), format(request.getEndTime()), request.getDescription());
+    }
+
+    @Before("bean(getExamsUseCase)")
+    public void logGetExamsUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var examFilter = (ExamFilter) args[0];
+        GET_EXAMS_USE_CASE_LOGGER.info("[Get Exams] studentId={} status=\"{}\" skip={} size={}",
+                examFilter.getStudentId(), examFilter.getStatus(), examFilter.getSkip(), examFilter.getSize());
+    }
+
+    @Before("bean(getExamProgressOverviewUseCase)")
+    public void logGetExamProgressOverviewUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var request = (GetExamProgressOverviewUseCase.Request) args[0];
+        GET_EXAM_PROGRESS_OVERVIEW_USE_CASE_LOGGER.info("[Get ExamProgress Overview] examId={} studentId={}",
+                request.getStudentId(), request.getStudentId());
+    }
+
+    @Before("bean(getExamOverviewUseCase)")
+    public void logGetExamOverviewUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var examId = (Integer) args[0];
+        GET_EXAM_OVERVIEW_USE_CASE_LOGGER.info("[Get Exam Overview] examId={}", examId);
+    }
+
+    @Before("bean(updateExamUseCase)")
+    public void logUpdateExamUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var request = (UpdateExamUseCase.Request) args[0];
+        UPDATE_EXAM_USE_CASE_LOGGER.info("[Update Exam] examId={} name=\"{}\" startTime=\"{}\" endTime=\"{}\" description=\"{}\"",
+                request.getExamId(), request.getName(), format(request.getStartTime()), format(request.getEndTime()), request.getDescription());
+    }
+
+    @Before("bean(getExamUseCase)")
+    public void logGetExamUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var request = (GetExamUseCase.Request) args[0];
+        GET_EXAM_USE_CASE_LOGGER.info("[Get Exam] examId={} onlyExamineeCanAccess={} studentId={}",
+                request.getExamId(), request.isOnlyExamineeCanAccess(), request.getStudentId());
+    }
+
+    @Before("bean(deleteExamineesUseCase)")
+    public void logDeleteExamineesUseCase(JoinPoint joinPoint) {
+        var args = joinPoint.getArgs();
+        var request = (DeleteExamineesUseCase.Request) args[0];
+        DELETE_EXAMINEES_USE_CASE_LOGGER.info("[Delete Examinees] examId={} emails={}",
+                request.getExamId(), request.getEmails());
```

Comment:
`String.join(", ", request.getEmails())`. static import

with it enclosed in a pair of `"`